presto-hive/src/main/java/com/facebook/presto/hive/HiveCoercer.java

```diff
@@ -81,10 +81,10 @@ static HiveCoercer createCoercer(TypeManager typeManager, HiveType fromHiveType,
         else if (fromType instanceof VarcharType && (toHiveType.equals(HIVE_BYTE) || toHiveType.equals(HIVE_SHORT) || toHiveType.equals(HIVE_INT) || toHiveType.equals(HIVE_LONG))) {
             return new VarcharToIntegerNumberCoercer(fromType, toType);
         }
-        else if (fromHiveType.equals(HIVE_BYTE) && toHiveType.equals(HIVE_SHORT) || toHiveType.equals(HIVE_INT) || toHiveType.equals(HIVE_LONG)) {
+        else if (fromHiveType.equals(HIVE_BYTE) && (toHiveType.equals(HIVE_SHORT) || toHiveType.equals(HIVE_INT) || toHiveType.equals(HIVE_LONG))) {
```

Comment:
@elharo Thanks for the fix. It does seem the original code has a bug. Will you be able to add some tests in com/facebook/presto/hive/TestCoercingFilters.java?

Reply:
done